ext/datasource/etcdv3/etcdv3.go

```diff
@@ -0,0 +1,112 @@
+package etcdv3
+
+import (
+	"context"
+	"github.com/alibaba/sentinel-golang/ext/datasource"
+	"github.com/alibaba/sentinel-golang/logging"
+	"github.com/alibaba/sentinel-golang/util"
+	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/mvcc/mvccpb"
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+	"time"
+)
+
+var logger = logging.GetDefaultLogger()
+
+type Etcdv3DataSource struct {
+	datasource.Base
+	client         *clientv3.Client
+	propertyKey    string
+	watchIndex     int64
```

Comment:
Maybe rename to 'revision' makes more sense.

Reply:
revision is the concept in etcd. Watch may be better here.